generate_parameter_library/cmake/generate_parameter_library.cmake

```diff
@@ -89,3 +89,36 @@ function(generate_parameter_library LIB_NAME YAML_FILE)
   )
   install(DIRECTORY ${LIB_INCLUDE_DIR} DESTINATION include/)
 endfunction()
+
+# create custom test function to pass yaml file into test main
+function(add_test_with_yaml_input_gtest TARGET SOURCES YAML_FILE)
```

Comment:
Should this be `add_rostest_with_parameters_gtest` or something to make it clear that the input yaml is for parameters and that these are a ros-test in that they only work if your test has ros in it (`rclcpp::init(argc, argv);`)?